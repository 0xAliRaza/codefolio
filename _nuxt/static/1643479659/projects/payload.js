__NUXT_JSONP__("/projects", (function(a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q){return {data:[{projects:[{slug:"codefolio",id:4,title:"Codefolio",description:"Lorem ipsum dolor, sit amet consectetur adipisicing elit. Alias, blanditiis! Ut nemo minus illo. Perferendis non adipisci harum voluptatem quam accusantium totam iure aperiam quasi. Perferendis odit ullam hic sed!",image:d,links:{liveDemo:a,videoDemo:a,sourceCode:a},technologies:[e,g,h,i,k,j,g,h,i,k,j],dir:f,path:"\u002Fprojects\u002Fcodefolio",extension:c,createdAt:b,updatedAt:b},{slug:"habitly",id:1,title:"Habitly",description:"Lorem ipsum dolor, sit amet consectetur adipisicing elit. Consequuntur, numquam perspiciatis? Rem iste ad cum optio deserunt vero quos, obcaecati ut dicta, tempore earum porro a facere! Quidem, atque excepturi.",image:d,links:{liveDemo:a,videoDemo:a,sourceCode:a},technologies:[e,g,h,i,"VUEJS","Django",j,"Docker"],dir:f,path:"\u002Fprojects\u002Fhabitly",extension:c,createdAt:b,updatedAt:b},{slug:"wordsery",id:3,title:"Wordsery",description:"Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe nobis, doloremque soluta tempora nisi laudantium quam. A error odio facere laudantium nam delectus mollitia expedita voluptas porro, at officiis culpa!.",image:d,links:{liveDemo:a,videoDemo:a,sourceCode:a},technologies:[e,l,m,n,"PHP",o,"vuejs",p,q],dir:f,path:"\u002Fprojects\u002Fwordsery",extension:c,createdAt:b,updatedAt:b},{slug:"write-it-down",id:2,title:"Write it down",description:"Lorem ipsum dolor sit amet consectetur adipisicing elit. Nihil in autem alias sapiente accusamus officiis facere, iusto dolor. Laborum consectetur voluptatem hic incidunt sit sequi saepe ullam in sapiente impedit.",image:d,links:{liveDemo:a,demoVideo:a,sourceCode:a},technologies:[e,l,m,n,"angular",o,p,q],dir:f,path:"\u002Fprojects\u002Fwrite-it-down",extension:c,createdAt:b,updatedAt:b}],navbar:{slug:"navbar",resumeUrl:"https:\u002F\u002Fexample.com\u002Fresume",externalLinks:[{text:"github",url:a},{text:"blog",url:"https:\u002F\u002Fexample.com\u002Fblog"}],dir:"\u002F",path:"\u002Fnavbar",extension:c,createdAt:b,updatedAt:b}}],fetch:{},mutations:void 0}}("#","2022-01-29T18:06:39.596Z",".json","https:\u002F\u002Fsource.unsplash.com\u002Frandom\u002F1920x1080","HTML","\u002Fprojects","CSS","SCSS","JavaScript","Git","NuxtJS","css","scss","javaScript","laravel","git","docker")));