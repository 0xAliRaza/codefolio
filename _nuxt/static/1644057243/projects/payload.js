__NUXT_JSONP__("/projects", (function(a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q){return {data:[{projects:[{slug:"habitly",id:1,title:"Habitly",description:"Lorem ipsum dolor, sit amet consectetur adipisicing elit. Consequuntur, numquam perspiciatis? Rem iste ad cum optio deserunt vero quos, obcaecati ut dicta, tempore earum porro a facere! Quidem, atque excepturi.",image:f,links:{liveDemo:a,videoDemo:a,sourceCode:a},technologies:[d,g,h,i,"VUEJS","Django",j,"Docker"],dir:e,path:"\u002Fprojects\u002Fhabitly",extension:c,createdAt:b,updatedAt:b},{slug:"write-it-down",id:2,title:"Write it down",description:"Lorem ipsum dolor sit amet consectetur adipisicing elit. Nihil in autem alias sapiente accusamus officiis facere, iusto dolor. Laborum consectetur voluptatem hic incidunt sit sequi saepe ullam in sapiente impedit.",image:f,links:{liveDemo:a,demoVideo:a,sourceCode:a},technologies:[d,k,l,m,"angular",n,o,p],dir:e,path:"\u002Fprojects\u002Fwrite-it-down",extension:c,createdAt:b,updatedAt:b},{slug:"wordsery",id:3,title:"Wordsery",description:"Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe nobis, doloremque soluta tempora nisi laudantium quam. A error odio facere laudantium nam delectus mollitia expedita voluptas porro, at officiis culpa!.",image:f,links:{liveDemo:a,videoDemo:a,sourceCode:a},technologies:[d,k,l,m,"PHP",n,"vuejs",o,p],dir:e,path:"\u002Fprojects\u002Fwordsery",extension:c,createdAt:b,updatedAt:b},{slug:"codefolio",id:4,title:"Codefolio",description:"Lorem ipsum dolor, sit amet consectetur adipisicing elit. Alias, blanditiis! Ut nemo minus illo. Perferendis non adipisci harum voluptatem quam accusantium totam iure aperiam quasi. Perferendis odit ullam hic sed!",image:"codefolio.jpeg",links:{liveDemo:a,videoDemo:a,sourceCode:a},technologies:[d,g,h,i,q,j,g,h,i,q,j],dir:e,path:"\u002Fprojects\u002Fcodefolio",extension:c,createdAt:b,updatedAt:b}],navbar:{slug:"navbar",resumeUrl:"https:\u002F\u002Fexample.com\u002Fresume",externalLinks:[{text:"github",url:a},{text:"blog",url:"https:\u002F\u002Fexample.com\u002Fblog"}],dir:"\u002F",path:"\u002Fnavbar",extension:c,createdAt:b,updatedAt:b}}],fetch:{},mutations:void 0}}("#","2022-02-05T10:33:10.586Z",".json","HTML","\u002Fprojects","https:\u002F\u002Fsource.unsplash.com\u002Frandom\u002F1920x1080","CSS","SCSS","JavaScript","Git","css","scss","javaScript","laravel","git","docker","NuxtJS")));