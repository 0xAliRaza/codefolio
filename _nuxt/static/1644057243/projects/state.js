window.__NUXT__=(function(a){return {staticAssetsBase:"\u002Fcodefolio\u002F_nuxt\u002Fstatic\u002F1644057243",layout:"default",error:a,serverRendered:true,routePath:"\u002Fprojects",config:{_app:{basePath:"\u002Fcodefolio\u002F",assetsPath:"\u002Fcodefolio\u002F_nuxt\u002F",cdnURL:a},content:{dbHash:"5250243d"}}}}(null));