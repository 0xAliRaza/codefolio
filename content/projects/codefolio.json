{
  "id": 4,
  "title": "Codefolio",
  "description": "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Alias, blanditiis! Ut nemo minus illo. Perferendis non adipisci harum voluptatem quam accusantium totam iure aperiam quasi. Perferendis odit ullam hic sed!",
  "image": "codefolio.jpeg",
  "links": {
    "liveDemo": "#",
    "videoDemo": "#",
    "sourceCode": "#"
  },
  "technologies": [
    "HTML",
    "CSS",
    "SCSS",
    "JavaScript",
    "NuxtJS",
    "Git",
    "CSS",
    "SCSS",
    "JavaScript",
    "NuxtJS",
    "Git"
  ]
}
