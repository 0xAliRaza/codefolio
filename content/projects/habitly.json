{
  "id": 1,
  "title": "Habitly",
  "description": "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Consequuntur, numquam perspiciatis? Rem iste ad cum optio deserunt vero quos, obcaecati ut dicta, tempore earum porro a facere! Quidem, atque excepturi.",
  "image": "https://source.unsplash.com/random/1920x1080",
  "links": {
    "liveDemo": "#",
    "videoDemo": "#",
    "sourceCode": "#"
  },
  "technologies": [
    "HTML",
    "CSS",
    "SCSS",
    "JavaScript",
    "VUEJS",
    "Django",
    "Git",
    "Docker"
  ]
}
