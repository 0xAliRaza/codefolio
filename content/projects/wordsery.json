{
  "id": 3,
  "title": "Wordsery",
  "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Saepe nobis, doloremque soluta tempora nisi laudantium quam. A error odio facere laudantium nam delectus mollitia expedita voluptas porro, at officiis culpa!.",
  "image": "https://source.unsplash.com/random/1920x1080",
  "links": {
    "liveDemo": "#",
    "videoDemo": "#",
    "sourceCode": "#"
  },
  "technologies": [
    "HTML",
    "css",
    "scss",
    "javaScript",
    "PHP",
    "laravel",
    "vuejs",
    "git",
    "docker"
  ]
}
