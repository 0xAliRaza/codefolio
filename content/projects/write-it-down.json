{
  "id": 2,
  "title": "Write it down",
  "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Nihil in autem alias sapiente accusamus officiis facere, iusto dolor. Laborum consectetur voluptatem hic incidunt sit sequi saepe ullam in sapiente impedit.",
  "image": "https://source.unsplash.com/random/1920x1080",
  "links": {
    "liveDemo": "#",
    "demoVideo": "#",
    "sourceCode": "#"
  },
  "technologies": [
    "HTML",
    "css",
    "scss",
    "javaScript",
    "angular",
    "laravel",
    "git",
    "docker"
  ]
}
