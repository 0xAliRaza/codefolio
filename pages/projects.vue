<template>
  <div>
    <Header
      :resume-url="navbar.resumeUrl"
      :external-links="navbar.externalLinks"
    />
    <main class="main">
      <div class="container">
        <div class="row">
          <div class="col-12">
            <section class="projects">
              <div class="text-center my-3">
                <h1>Projects</h1>
              </div>
              <div class="row justify-content-center py-4">
                <div
                  v-for="project in projects"
                  :key="project.id"
                  class="col-12 col-md-10 col-lg-6 col-xl-5 mb-4 mb-lg-5"
                >
                  <Project
                    :title="project.title"
                    :description="project.description"
                    :image="project.image"
                    :links="project.links"
                    :technologies="project.technologies"
                  />
                </div>
              </div>
            </section>
          </div>
        </div>
      </div>
    </main>
    <Footer />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  async asyncData({ $content }) {
    const projects = (await $content('projects').fetch()).sort(
      // Sort by projects by id in ascending order
      (a: any, b: any) => {
        if (a.id > b.id) {
          return 1
        }
        if (a.id < b.id) {
          return -1
        }
        return 0
      }
    )
    const navbar = await $content('navbar').fetch()
    return {
      projects,
      navbar,
    }
  },
  head: {
    title: 'Projects - Codefolio',
  },
})
</script>

<style lang="scss" scoped>
@import '@/scss/abstracts';
.projects {
  padding: pxToRem(10) 0;
}
</style>
