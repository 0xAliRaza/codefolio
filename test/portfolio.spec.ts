import { shallowMount } from '@vue/test-utils'
import projects from '@/pages/projects.vue'

describe('projects', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(projects, {
      data() {
        return {
          navbar: { resumeUrl: '', externalLinks: [] },
        }
      },
      stubs: ['Icon'],
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
