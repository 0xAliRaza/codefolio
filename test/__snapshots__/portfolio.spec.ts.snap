// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`projects renders correctly 1`] = `
<div>
  <header resume-url
          external-links
  >
  </header>
  <main class="main">
    <div class="container">
      <div class="row">
        <div class="col-12">
          <section class="projects">
            <div class="text-center my-3">
              <h1>
                Projects
              </h1>
            </div>
            <div class="row justify-content-center py-4">
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
  <footer>
  </footer>
</div>
`;
